### PR TITLE
feat(web): US-20.5 Public Profile Page — /@handle (#217)

### LINKED ISSUE #217: US-20.5: Profile Page
## User Story

**As a** listener, **I want** to view a musician's public profile showing their avatar, bio, published songs, playlists, and follower counts, **so that** I can explore their work and follow them.

## Description

Build the public profile page at `/@username` from spec section 32. Displays the creator's identity, published content, and social connections.

## Functional Requirements

- Avatar, display name, and bio prominently displayed
- Style tags rendered as pill badges
- Published songs grid (paginated)
- Playlists section
- Follower and following counts
- Follow/unfollow button (for authenticated users viewing others' profiles)
- Tab navigation between songs, playlists, and about sections
- Profile URL matches the user's handle (`/@username`)

## Acceptance Criteria

- [ ] Profile page loads at `/@username` with correct user data
- [ ] Published songs render in a grid with clip cards
- [ ] Follow button toggles follow state and updates follower count
- [ ] Style tags render as pill badges
- [ ] Playlists section shows the user's public playlists

## Dependencies

**Blocked by:** US-20.4 (prior story in this stage). Part of Stage 20: Discovery & Social UI, which is blocked by Stage 19.

## References

- User stories: `user-stories/03-layer-3-web-ui.md` — Stage 20, US-20.5

### PR DESCRIPTION
**Pull request by frankbria**
## US-20.5 Public Profile Page (#217)

Public creator profile at `/@handle` — Stage 20: Discovery & Social UI.

### Approach: web-only against `lib/*` mock seams

Following the established Stage-20 precedent (US-20.1 Explore #304, US-20.2 Search #305, US-20.3 Playlists #306, US-20.4 Feed #307), this ships **web-only**. The issue's auto-generated CodeRabbit plan proposed a `Follow` Beanie collection + three FastAPI endpoints; that backend doesn't exist yet, so — like every prior Stage-20 story — the page runs on a typed mock seam whose shapes mirror the eventual public profile/social API. Each getter swaps its body for a `fetch` when the backend lands; callers stay unchanged.

### Changes

| File | What |
|------|------|
| `web/lib/profiles.ts` | `PublicProfile` seam: `getProfileByHandle` (handle normalization), `profileClips` (resolved against the Explore pool), `publicPlaylists` (public seed playlists). |
| `web/app/[handle]/page.tsx` | Root dynamic route. Static routes win over it; `notFound()` on non-`@` or unknown handles so it can't hijack real paths. Decodes the raw `useParams` segment. |
| `web/components/profile/ProfileView.tsx` | Header (glyph avatar, name, `@handle`, bio, style pills, follower/following) + Songs / Playlists / About tabs. |
| `web/components/profile/FollowButton.tsx` | Controlled follow toggle (`aria-pressed`), optimistic ±1 follower count. |
| `+ profiles.test.ts, ProfileView.test.tsx, app/[handle]/page.test.tsx` | 14 tests. |

### Acceptance criteria

- [x] Profile loads at `/@username` with correct data → route + `getProfileByHandle`
- [x] Published songs render in a grid of clip cards → Songs tab × `ExploreClipCard`
- [x] Follow button toggles state + updates follower count → `FollowButton` (optimistic ±1)
- [x] Style tags render as pill badges → `Badge` row (header + About)
- [x] Playlists section shows public playlists → Playlists tab × `publicPlaylists`

### Reuse (no new deps)

`ExploreClipCard`, `ui/{tabs,badge,button}`, `useAuth`, `lib/playlists` seed data.

### Verification

- `tsc --noEmit` clean · `eslint` clean · **1239/1239** existing tests + **14** new · `next build` OK (`[handle]` route registered)
- Cross-family review (**codex**): no findings.
- Internal review: 0 Critical / 0 Major; 1 Minor (route-guard test) + 1 Nit (aria-pressed re-check) **fixed** in follow-up commit; 1 Nit (always-plural labels) declined — matches Twitter/GitHub convention.

### Known limitations (deferred)

- No real Follow backend — follow state is **session-only** (optimistic, resets on reload), matching the Stage-20 mock-seam pattern.
- No follower/following **lists** or deep pagination — a single songs grid.
- **No self-view guard** — the Follow button shows for any authenticated viewer because the session user (`AuthUser = {id, email}`) carries no handle yet; add the guard when the token exposes a handle.
- Avatar is an initials/glyph placeholder — there's no authed artwork proxy.

Closes #217.

### CODE REVIEW

**Comment by frankbria** [comment]
## 🎬 Demo verification (Phase 11)

Ran the feature on a live `next dev` server + the interaction suite. Every acceptance criterion has outcome evidence.

| AC | Evidence | Result |
|----|----------|--------|
| **1** Profile loads at `/@username` with correct data | `GET /@nova → 200`, HTML contains "Nova Bloom", "@nova", bio text | ✅ |
| **2** Published songs render in a grid of clip cards | HTML contains "Neon Skyline", "Pulse Theory", "Paper Lanterns" (3 Explore cards) | ✅ |
| **3** Follow toggles state + updates follower count | `ProfileView` test: click → `Following` + `aria-pressed=true` + count title `12,800 → 12,801`; click again → back to `12,800` | ✅ |
| **4** Style tags render as pill badges | HTML `Badge`s: synthwave, electronic, ambient | ✅ |
| **5** Playlists section shows public playlists | test: Playlists tab shows "Late Night Drive"/"Summer Anthems"; private "Deep Focus" **absent** | ✅ |

**Routing edge cases (live HTTP):** `/@Nova` (case) → 200 · `/@ghostperson` (unknown) → 404 · `/random-typo` (non-@) → 404 · `/explore` (static route not hijacked) → 200.

**Follow-button gating:** absent from anonymous SSR; shown only to authenticated viewers.

Suite: **14 new tests pass** (6 ProfileView + 4 route shim + 4 seam) + 1239/1239 existing. `tsc`/`eslint`/`next build` clean. Cross-family (codex) review: no findings.

**Comment by frankbria** [comment]
## Review triage (claude-review)

Thanks — traced each finding, applied the cheap correctness/a11y ones in `faeb78f`:

| # | Finding | Action |
|---|---------|--------|
| 1 | `decodeURIComponent` can throw on a malformed catch-all path | **Fixed** — wrapped in try/catch → `notFound()`. (Empirically Next 400s malformed URIs before the route runs, so it wasn't reachable today, but a public root catch-all is a trust boundary worth hardening unconditionally.) + test for `%zz`. |
| 2 | Follow state could leak across a same-instance `/@a`→`/@b` nav (latent, no such links today) | **Fixed** — `key={handle}` forces a clean remount per profile. |
| 3 | Exact count only in `title` (hover-only, not SR-announced) | **Fixed** — added `sr-only` full number; compact form is now `aria-hidden`. |
| 4 | No dedicated `FollowButton` test | **Declined** — presentational, covered via `ProfileView.test.tsx`; matches the `SectionRow`/`GenreChannels` precedent from US-20.1. |
| 5 | No SSR `generateMetadata`/OG tags | **Deferred** — the page is client-only like every Stage-20 surface; adding metadata needs a server-wrapper restructure. Tracked with the other backend-phase deferrals in the PR's Known Limitations. |

Tests now 15 (added malformed-handle 404). `tsc`/`eslint` clean.

File: web/app/[handle]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+// Public profile route /@handle (US-20.5). This is a root dynamic segment, so it
+// also catches unknown top-level paths — static routes (/explore, /search, …) win
+// over it, and anything reaching here that isn't "@handle" is a 404. The "@" is
+// part of the captured param; ProfileView strips it and 404s on an unknown handle.
+
+export default function ProfilePage() {
+  const params = useParams<{ handle: string }>()
+  const handle = params?.handle ? decodeURIComponent(params.handle) : ""
+  if (!handle.startsWith("@")) notFound()
```
[the remark this answers is not in the document]
Comment (frankbria) [reply web/app/[handle]/page.tsx]: Fixed in `faeb78f` — this finding predates that commit. `decodeHandle()` now wraps the decode in try/catch and returns `""` on a bad escape, which fails the `startsWith("@")` check → clean `notFound()`. New test covers `%zz`. (Keeping the manual decode: `useParams()` was verified to hand over the raw, still-encoded segment, so it's needed for canonical `/@handle` resolution — and it's safe either way.)
